src/test/kotlin/org/rust/lang/refactoring/RsExtractFunctionTest.kt

```diff
@@ -33,6 +33,48 @@ class RsExtractFunctionTest : RsTestBase() {
         false,
         "test")
 
+    fun `test extract basic return type`() = doTest("""
+            fn main() {
+                <selection>let test = 10i32;</selection>
+                println!("{}", test);
+            }
+        """, """
+            fn main() {
+                let test = test();
+                println!("{}", test);
+            }
+
+            fn test() -> i32 {
+                let test = 10i32;
+                test
+            }
+        """,
+        false,
+        "test")
+
+    fun `test extract take last possible return binding`() = doTest("""
+            fn main() {
+                <selection>let test2 = 10i32;
+                let test = 10i32;</selection>
+                println!("{}", test);
+                println!("{}", test2);
+            }
+        """, """
+            fn main() {
+                let test = test();
```

Comment:
Yeah, ideally, we should be able to do something like `let (test, test2) = test()`...